src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseCancellationTokenThrowIfCancellationRequested.cs

```diff
@@ -0,0 +1,193 @@
+                IPropertySymbol? isCancellationRequestedProperty = cancellationTokenType.GetMembers(nameof(CancellationToken.IsCancellationRequested))
+                    .OfType<IPropertySymbol>()
+                    .FirstOrDefault();
+
+                if (throwIfCancellationRequestedMethod is null || isCancellationRequestedProperty is null)
+                    return false;
+
+                if (!compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemOperationCanceledException, out INamedTypeSymbol? operationCanceledExceptionType))
+                    return false;
+
+                IMethodSymbol? operationCanceledExceptionDefaultCtor = operationCanceledExceptionType.InstanceConstructors
+                    .GetFirstOrDefaultMemberWithParameterInfos();
+                IMethodSymbol? operationCanceledExceptionTokenCtor = operationCanceledExceptionType.InstanceConstructors
+                    .GetFirstOrDefaultMemberWithParameterInfos(ParameterInfo.GetParameterInfo(cancellationTokenType));
+
+                if (operationCanceledExceptionDefaultCtor is null || operationCanceledExceptionTokenCtor is null)
+                    return false;
+
+                symbols = new RequiredSymbols
+                {
+                    IsCancellationRequestedProperty = isCancellationRequestedProperty,
+                    OperationCanceledExceptionDefaultCtor = operationCanceledExceptionDefaultCtor,
+                    OperationCanceledExceptionTokenCtor = operationCanceledExceptionTokenCtor
+                };
+
+                return true;
+            }
+
+            public IPropertySymbol IsCancellationRequestedProperty { get; init; }
+            public IMethodSymbol OperationCanceledExceptionDefaultCtor { get; init; }
+            public IMethodSymbol OperationCanceledExceptionTokenCtor { get; init; }
+
+            /// <summary>
+            /// Indicates whether the specified operation is a conditional statement of the form
+            /// <code>
+            /// if (token.IsCancellationRequested)
+            ///     throw new OperationCanceledException();
+            /// </code>
+            /// </summary>
+            public bool IsSimpleAffirmativeCheck(IConditionalOperation conditional, [NotNullWhen(true)] out IPropertyReferenceOperation? isCancellationRequestedPropertyReference)
+            {
+                IOperation? whenTrueUnwrapped = GetSingleStatementOrDefault(conditional.WhenTrue);
+
+                if (conditional.Condition is IPropertyReferenceOperation propertyReference &&
+                    SymbolEqualityComparer.Default.Equals(propertyReference.Property, IsCancellationRequestedProperty) &&
+                    whenTrueUnwrapped is IThrowOperation @throw &&
+                    @throw.Exception is IObjectCreationOperation objectCreation &&
+                    IsDefaultOrTokenOperationCanceledExceptionCtor(objectCreation.Constructor) &&
+                    conditional.WhenFalse is null)
```

Comment:
Why this restriction? I believe the code can still be written as `cancellationToken.ThrowIfCancellationRequested();` followed by the body of the else/WhenFalse block.

Reply:
I was implementing the analyzer as specified [here](https://github.com/dotnet/runtime/issues/33809#issuecomment-761146782), which specified two cases that should be detected and fixed:
1. A simple affirmative check with no else clause, or
2. A negated check with an else clause. 
I'll remove the restriction.